zebrad/src/components/inbound.rs

```diff
@@ -50,6 +52,12 @@ mod tests;
 
 use downloads::Downloads as BlockDownloads;
 
+/// The maximum amount of time an inbound service response can take.
+///
+/// If the response takes longer than this time, it will be cancelled,
+/// and the peer might be disconnected.
+pub const MAX_INBOUND_RESPONSE_TIME: Duration = Duration::from_secs(5);
```

Comment:
Our outbound request timeout is 20 seconds, but that seems like a long time to block inbound requests.